scripts/scil_split_image.py

```diff
@@ -0,0 +1,147 @@
+
+"""
+Splits the DWI image at certain indices along the last dimension (b-values).
+Many indices can be given at once by specifying multiple values. The splited
+volumes are in the same order as in the original file. Also outputs the
+corresponding .bval and .bvec files.
+
+This script can be useful for splitting images at places where a b-value
+extraction does not work. For instance, if one wants to split the x first
+b-1500s from the rest of the b-1500s in an image, simply put x as an index.
+
+"""
+
+import argparse
+import logging
+from pathlib import Path
+
+from dipy.io import read_bvals_bvecs
+import nibabel as nib
+import numpy as np
+
+from scilpy.io.utils import (add_overwrite_arg, add_verbose_arg,
+                             assert_inputs_exist, assert_outputs_exist)
+
+
+def _build_arg_parser():
+    p = argparse.ArgumentParser(
+        description=__doc__,
+        formatter_class=argparse.RawTextHelpFormatter)
+
+    p.add_argument('in_dwi',
+                   help='The DW image file to split.')
+
+    p.add_argument('in_bval',
+                   help='The b-values file in FSL format (.bval).')
+
+    p.add_argument('in_bvec',
+                   help='The b-vectors file in FSL format (.bvec).')
+
+    p.add_argument('split_indices', nargs='+', type=int,
+                   help='The list of indices where to split the image. For '
+                        'example 3 10. This would split the image in three '
+                        'parts, such as [:3], [3:10], [10:].')
+
+    p.add_argument('--out_dwi', nargs='+', default=[],
+                   help='The names of the output DWI files. There must be '
+                        'one more name than indices in split_indices. By '
+                        'default, indices number will be appended to in_dwi, '
+                        'such as in_dwi_0_3, in_dwi_3_10, in_dwi_10_end.')
+
+    p.add_argument('--out_bval', nargs='+', default=[],
+                   help='The names of the output b-value files (.bval). There '
+                        'must be one more name than indices in split_indices. '
+                        'By default, indices number will be appended to '
+                        'in_dwi, such as in_dwi_0_3, in_dwi_3_10, '
+                        'in_dwi_10_end.')
+
+    p.add_argument('--out_bvec', nargs='+', default=[],
+                   help='The names of the output b-vector files (.bvec).There '
+                        'must be one more name than indices in split_indices. '
+                        'By default, indices number will be appended to '
+                        'in_dwi, such as in_dwi_0_3, in_dwi_3_10, '
+                        'in_dwi_10_end.')
+
+    add_verbose_arg(p)
+    add_overwrite_arg(p)
+
+    return p
+
+
+def create_name_from_indices(indices, input):
+    input_path = Path(input)
+    parent_dir = input_path.parent
+    input_name = Path(input_path.stem).stem
+    index_name = "_" + str(indices[0]) + "_" + str(indices[1])
```

Comment:
Why are you calling this function with a indices[i:i+2]?

Also should indices[1] be [-1] to capture the first and last?